src/data/list/basic.lean

```diff
@@ -17,6 +17,16 @@ variables {ι : Type*} {α : Type u} {β : Type v} {γ : Type w} {δ : Type x}
 
 attribute [inline] list.head
 
+-- TODO[gh-6025]: make this an instance once safe to do so
+/-- There is only one list of an empty type -/
+def unique_of_is_empty [is_empty α] : unique (list α) :=
+{ uniq := begin
+    rintro (_ | ⟨a, l⟩),
+    { refl },
+    { exact is_empty_elim a}
+  end
```

Comment:
Maybe nicer as
```suggestion
{ uniq := λ l, match l with
    | [] := rfl
    | (a :: l) := is_empty_elim a
    end,
```
I couldn't get `list.rec` to elaborate nicely.

Up to you which you prefer